subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -1044,6 +1044,40 @@ int bt_mesh_cfg_net_key_get(uint16_t net_idx, uint16_t addr, uint16_t *keys,
 	return cli_wait();
 }
 
+int bt_mesh_cfg_net_key_del(uint16_t net_idx, uint16_t addr,
+			    uint16_t key_net_idx, uint8_t *status)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_NET_KEY_DEL, 2);
+	struct bt_mesh_msg_ctx ctx = {
+		.net_idx = net_idx,
+		.app_idx = BT_MESH_KEY_DEV_REMOTE,
+		.addr = addr,
+		.send_ttl = BT_MESH_TTL_DEFAULT,
+	};
+	struct net_key_param param = {
+		.status = status,
+		.net_idx = key_net_idx,
+	};
+	int err;
+
+	err = cli_prepare(&param, OP_NET_KEY_STATUS);
+	if (err) {
+		return err;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_NET_KEY_DEL);
+	net_buf_simple_add_le16(&msg, key_net_idx);
+
+	err = bt_mesh_model_send(cli->model, &ctx, &msg, NULL, NULL);
+	if (err) {
+		BT_ERR("model_send() failed (err %d)", err);
+		cli_reset();
+		return err;
+	}
```

Comment:
Question if this part should be here for this function;
```
	if (!status) {
		cli_reset();
		return 0;
	}
```

Reply:
IIRC that's been used for other APIs to allow not waiting for the status, i.e. do the call asynchronously without a chance to know if it succeeded or not. I don't have a particularly strong opinion on this/

Reply:
Well spotted. Seems like this function should allow non-blocking operation, so I added it. There are a couple of other setters that are missing this, I'll make a separate PR for those. I think it's only the getters that don't need this.